### ISSUE TITLE
Time varying state space matrices

### ISSUE THREAD

**Issue by codles** [ElOceanografo/StateSpace.jl#9]
Do you have any interest in allowing the matrices in the state space models to be time varying?

[TimeModels](https://github.com/JuliaStats/TimeModels.jl) accomplishes this using a generator function for each matrix ([initial implentation](https://github.com/JuliaStats/TimeModels.jl/pull/32), [optimised implementation](https://github.com/JuliaStats/TimeModels.jl/pull/52)).

**Comment by JonnyCBB** [comment]
I like this idea. I hadn't really thought about it too much before. Would it just be a case of defining a matrix of functions and evaluating the matrix at each time step? The bench marking looks good too. I thought there would've been more overhead.

**Comment by ElOceanografo** [comment]
Yes, definitely.  Do you have any interest in porting your implementation over to StateSpace?

**Comment by codles** [comment]
Glad you are interested. I am happy to implement this. But won't be able to for another two weeks.

**Comment by JonnyCBB** [comment]
Fantastic :+1:

**Comment by ElOceanografo** [comment]
This is tentatively supported now ( e1b867a0fbe2ac90ca5f7aaedd50d7b48fec8c1c ) for `LinearGaussianSSM` .  Basically a copy-and-paste of the TimeModels implementation, without the feed-forward matrix, at least for now.

We should also think about what time-dependent behavior we might want for the nonlinear model types, and how to make this easy and consistent to specify for the end user...

**Comment by codles** [comment]
I have been using your dev branch implementation of time dependent behaviour for the last few days and it works a treat.
